anomalib/models/csflow/torch_model.py

```diff
@@ -0,0 +1,530 @@
+            in_channels,
+            channels_hidden,
+            kernel_size=kernel_size,
+            padding=pad,
+            bias=not batch_norm,
+            padding_mode=pad_mode,
+        )
+        self.conv_scale0_1 = nn.Conv2d(
+            channels_hidden * 1,
+            channels,  #
+            kernel_size=kernel_size,
+            padding=pad,
+            bias=not batch_norm,
+            padding_mode=pad_mode,
+            dilation=1,
+        )
+        self.conv_scale1_1 = nn.Conv2d(
+            channels_hidden * 1,
+            channels,  #
+            kernel_size=kernel_size,
+            padding=pad * 1,
+            bias=not batch_norm,
+            padding_mode=pad_mode,
+            dilation=1,
+        )
+        self.conv_scale2_1 = nn.Conv2d(
+            channels_hidden * 1,
+            channels,  #
+            kernel_size=kernel_size,
+            padding=pad,
+            bias=not batch_norm,
+            padding_mode=pad_mode,
+        )
+
+        self.upsample = nn.Upsample(scale_factor=2, mode="bilinear", align_corners=False)
+
+        self.up_conv10 = nn.Conv2d(
+            channels_hidden, channels, kernel_size=kernel_size, padding=pad, bias=True, padding_mode=pad_mode
+        )
+
+        self.up_conv21 = nn.Conv2d(
+            channels_hidden, channels, kernel_size=kernel_size, padding=pad, bias=True, padding_mode=pad_mode
+        )
+
+        self.down_conv01 = nn.Conv2d(
+            channels_hidden,
+            channels,
+            kernel_size=kernel_size,
+            padding=pad,
+            bias=not batch_norm,
+            stride=2,
+            padding_mode=pad_mode,
+            dilation=1,
+        )
+
+        self.down_conv12 = nn.Conv2d(
+            channels_hidden,
+            channels,
+            kernel_size=kernel_size,
+            padding=pad,
+            bias=not batch_norm,
+            stride=2,
+            padding_mode=pad_mode,
+            dilation=1,
+        )
+
+        self.leaky_relu = nn.LeakyReLU(self.leaky_slope)
+
+    def forward(self, scale0, scale1, scale2):
+        """Applies the cross convolution to the three scales."""
+        out0 = self.conv_scale0_0(scale0)
+        out1 = self.conv_scale1_0(scale1)
+        out2 = self.conv_scale2_0(scale2)
+
+        lr0 = self.leaky_relu(out0)
+        lr1 = self.leaky_relu(out1)
+        lr3 = self.leaky_relu(out2)
+
+        out0 = self.conv_scale0_1(lr0)
+        out1 = self.conv_scale1_1(lr1)
+        out2 = self.conv_scale2_1(lr3)
+
+        y1_up = self.up_conv10(self.upsample(lr1))
+        y2_up = self.up_conv21(self.upsample(lr3))
+
+        y0_down = self.down_conv01(lr0)
+        y1_down = self.down_conv12(lr1)
+
+        out0 = out0 + y1_up
+        out1 = out1 + y0_down + y2_up
+        out2 = out2 + y1_down
+
+        if self.use_gamma:
+            out0 = out0 * self.gamma0
+            out1 = out1 * self.gamma1
+            out2 = out2 * self.gamma2
+        return out0, out1, out2
```

Comment:
This foward method is a bit hard to follow. Do you think it could be simplified by grouping some of the computations? For example, the Conv2D -> LeakyReLU -> Conv2D could be grouped into a sequential block or nn.Module class maybe? Otherwise it would probably suffice to add some comments in the `__init__` and `__forward__` methods.

Reply:
I have added some comments but the problem is that since this is cross-scale, it performs computation on multiple inputs and returns multiple outputs. I am not sure if we can wrap some subset of the blocks here in a sequential module. But this is basically what this block does.
![image](https://user-images.githubusercontent.com/17232914/207386351-a1d4dcf7-14c3-433c-a074-69fa239374e6.png)